### PR TITLE
Update DeviceOs to latest protocol

### PR DESCRIPTION
**Pull request by rtm516**
Updated to follow the newly published protocol specification. Will probably need a floodgate update aswell.

https://github.com/CloudburstMC/protocol-docs/blob/master/enums/BuildPlatform.md

We likely want to merge https://github.com/GeyserMC/api/pull/4 first

### CODE REVIEW

**Review by onebeastchris — APPROVED** [review]
Yeah, this will likely need a new Floodgate build

**Review by Redned235 — APPROVED** [review]

File: common/src/main/java/org/geysermc/floodgate/util/DeviceOs.java [hunk 1]
```diff
@@ -39,15 +39,16 @@ public enum DeviceOs {
     OSX("macOS"),
     AMAZON("Amazon"),
     GEARVR("Gear VR"),
-    HOLOLENS("Hololens"),
+    HOLOLENS("Hololens"), // Removed from protocol but kept to keep numbers the same
```
Comment (Camotoy) [inline common/src/main/java/org/geysermc/floodgate/util/DeviceOs.java:42]: Mark this as deprecated instead?

File: common/src/main/java/org/geysermc/floodgate/util/DeviceOs.java [hunk 2]
```diff
@@ -39,15 +39,16 @@ public enum DeviceOs {
...
     UWP("Windows"),
     WIN32("Windows x86"),
     DEDICATED("Dedicated"),
     TVOS("Apple TV"),
-    PS4("PS4"),
+    SONY("PlayStation"),
```
Comment (Camotoy) [inline common/src/main/java/org/geysermc/floodgate/util/DeviceOs.java:47]: Just add a Javadoc indicating this isn't just for PS4. We can change the string though.

**Review by Camotoy — CHANGES_REQUESTED** [review]

**Review by Camotoy — APPROVED** [review]

**Review by Tim203 — CHANGES_REQUESTED** [review]
I wouldn't mark them as deprecated instead I'd add it in a comment saying that those platforms are no longer in use.
Deprecated would indicate to me that they will be removed in the future, but I don't think they'd do that because it'd shift the enum values.

**Comment by rtm516** [comment]
> I wouldn't mark them as deprecated instead I'd add it in a comment saying that those platforms are no longer in use. Deprecated would indicate to me that they will be removed in the future, but I don't think they'd do that because it'd shift the enum values.

They removed Hololens from the docs but left the gap in the enum values so I think depricated is good, for this. They also marked the others as depricted. If they add more they could replace the old values.

**Review by Tim203 — APPROVED** [review]

**Pull request merged**